addons/calendar_bis/models/calendar_timeslot.py

```diff
@@ -0,0 +1,234 @@
+    monthday = fields.Integer(compute="_compute_recurring", inverse="_inverse_recurring")
+    monthweekday_n = fields.Integer(compute="_compute_recurring", inverse="_inverse_recurring")
+    monthweekday_day = fields.Selection([('mo', 'Monday'), ('tu', 'Tuesday'), ('we', 'Wednesday'),
+        ('th', 'Thursday'), ('fr', 'Friday'), ('sa', 'Saturday'), ('su', 'Sunday')], compute="_compute_recurring", inverse="_inverse_recurring")
+
+    def write(self, values):
+        edit = values.pop('edit', 'all')
+        batch = self.env['calendar.timeslot']
+        if 'start' in values and isinstance(values['start'], str):
+            values['start'] = datetime.fromisoformat(values['start'])
+        if 'stop' in values and isinstance(values['stop'], str):
+            values['stop'] = datetime.fromisoformat(values['stop'])
+        if 'event_id' in values:
+            return super().write(values)
+        for slot in self:
+            if not slot.is_recurring:
+                batch += slot
+            elif edit == 'one':
+                slot.event_id.exdate(slot)
+                batch += slot
+            elif edit in ['post', 'all']:
+                if edit == 'post':
+                    slot.event_id.break_after(slot)
+                if 'start' not in values and 'stop' not in values:
+                    batch += slot.event_id.timeslot_ids
+                else:
+                    start = values.get('start', slot.start)
+                    start_delta = start - slot.start
+                    stop = values.get('stop', slot.stop)
+                    duration = max(values.get('duration', (stop - start).total_seconds() / 3600), 1/60)
+                    new_vals = {**values.copy(), 'duration': duration}
+                    for slot in slot.event_id.timeslot_ids:
+                        if start_delta: new_vals['start'] = slot.start + start_delta
+                        super(CalendarTimeslot, slot).write(new_vals)
+                        # TODO set attendee status to ???
+                    slot.update_recurrence_start(start, start_delta)
+        return super(CalendarTimeslot, batch).write(values)
+
+    @api.model_create_multi
+    def create(self, values):
+        res = super().create(values)
+        for slot, vals in zip(res, values):
+            if not 'partner_ids' in vals and not 'attendee_mail_ids' in vals:
+                continue
+            slot.partner_ids = vals.get('partner_ids', [])
+            slot.attendee_mail_ids = vals.get('attendee_mail_ids', [])
+            attendee_pids = slot.attendee_ids.partner_id.ids
+            attendee_mails = slot.attendee_ids.mapped('email')
+            slot.attendee_ids += slot.attendee_ids.create(
+                [{'partner_id': pid.id, 'timeslot_id': slot.id} for pid in slot.partner_ids if pid not in attendee_pids] + \
+                [{'email': attmail.name, 'timeslot_id': slot.id} for attmail in slot.attendee_mail_ids if attmail not in attendee_mails])
```

Comment:
Should be validated (do we really want to do this), and probably moved into an independent PR.